pkgs/by-name/ma/magicrescue/package.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, gdbm # ndbm.h for dupemap binary
+, perl
+}:
+
+stdenv.mkDerivation {
+  pname = "magicrescue";
+  version = "1.1.10-unstable-2021-09-12";
+
+  src = fetchFromGitHub {
+    owner = "jbj";
+    repo = "magicrescue";
+    rev = "d9a57931d437674009bfd2f98451b3d71058eade";
+    hash = "sha256-jVBzsa39TQjeDZ4zuXn3UA+4WectjNRwPNb1AkLuIbg=";
+  };
+
+  patches = [
+    # Add PERL as processor for file.
+    (fetchpatch {
+      url = "https://salsa.debian.org/pkg-security-team/magicrescue/-/raw/6331d088a159ae21ad4ab5f18b9bf892ebe18ce3/debian/patches/020_add-Perl-preprocessor.patch";
+      hash = "sha256-XX3Rlv/qKB2y/csuaPiliv4cu9KKHNpG/E88VSVP0sg=";
+    })
+  ];
+
+  buildInputs = [
+    gdbm
+  ];
+
+  postPatch = ''
+    substituteInPlace magicsort \
+      --replace /usr/bin/perl ${lib.getExe perl}
+  '';
```

Comment:
```suggestion
  buildInputs = [
    gdbm
    perl
  ];
```

Reply:
@a-n-n-a-l-e-e by removing `substituteinPlace`, did you test that `magicsort` is correctly working on the `perl` part?

Reply:
https://nixos.org/manual/nixpkgs/unstable/#patch-shebangs.sh

`patchShebangs --host $out` is run in the fixupPhase and will update the shebang of executable scripts for programs installed on the host. thus putting perl in buildInputs allows patchShebangs to find perl and thus will patch the script.

however you should validate it works.